extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultHealthCheck.java

```diff
@@ -40,51 +45,27 @@ public HashicorpVaultHealthCheck(HashicorpVaultClient client, Monitor monitor) {
     @Override
     public HealthCheckResult get() {
 
-        HealthResponse response;
-        HealthCheckResult result;
-        try {
-            response = client.getHealth();
-        } catch (EdcException e) {  // can be thrown by the client, e.g. on JSON parsing error, etc.
-            var exceptionMsg = format(HEALTH_CHECK_ERROR_TEMPLATE, "EdcException: " + e.getMessage(), "");
-            monitor.severe(exceptionMsg, e);
-            return HealthCheckResult.failed(exceptionMsg).forComponent(COMPONENT_NAME);
+        var errors = new ArrayList<String>();
```

Comment:
to gather errors from multiple result the `merge` method can be used. actually this method could be implemented as:

```java
        return client.doHealthCheck().merge(client.lookUpToken())
                .flatMap(result -> {
                    if (result.succeeded()) {
                        return HealthCheckResult.success();
                    } else {
                        monitor.warning("Healthcheck failed with reason(s): " + result.getFailureDetail());
                        return HealthCheckResult.failed(result.getFailureMessages());
                    }
                }).forComponent(HashicorpVaultHealthExtension.NAME);
```